crates/pepper/src/catalog.rs

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//! Trait abstraction for metadata catalog operations.
+//!
+//! This trait defines the interface for managing table metadata
+//! and file references. It can be implemented by different RDBMS backends
+//! (`SQLite`, `PostgreSQL`, etc.).
+
+use super::metadata::{CreateTableOptions, DataFile, DeleteFile, TableMetadata, TableStats};
+use async_trait::async_trait;
+use snafu::Snafu;
+use std::sync::Arc;
+
+/// Error type for catalog operations.
+#[derive(Debug, Snafu)]
+pub enum CatalogError {
+    /// Database error
+    #[snafu(display("Database error: {message}"))]
+    Database {
+        /// Error message
+        message: String,
+    },
+
+    /// Table not found
+    #[snafu(display("Table not found: {table_name}"))]
+    TableNotFound {
+        /// Name of the table that was not found
+        table_name: String,
+    },
+
+    /// Table already exists
+    #[snafu(display("Table already exists: {table_name}"))]
+    TableAlreadyExists {
+        /// Name of the table that already exists
+        table_name: String,
+    },
+
+    /// Invalid operation
+    #[snafu(display("Invalid operation: {message}"))]
+    InvalidOperation {
+        /// Description of the invalid operation
+        message: String,
+    },
+
+    /// IO error
+    #[snafu(display("IO error: {source}"))]
+    Io {
+        /// The underlying IO error
+        source: std::io::Error,
+    },
+
+    /// `SQLite` error
+    #[snafu(display("SQLite error: {source}"))]
```

Comment:
```suggestion
    #[snafu(transparent)]
```

https://docs.rs/snafu/latest/snafu/derive.Snafu.html#delegating-to-the-underlying-error